### PR TITLE
Enabled data parallel for classifier-free inference

### PR DESCRIPTION
**Pull request by ExtremeViscent**

### CODE REVIEW

File: opendit/models/latte/latte_t2v.py [hunk 1]
```diff
@@ -1138,6 +1146,29 @@ def forward(
             If `return_dict` is True, an [`~models.transformer_2d.Transformer2DModelOutput`] is returned, otherwise a
             `tuple` where the first element is the sample tensor.
         """
+
+        # 0. Split batch for data parallelism
+        if get_data_parallel_size() > 1:
```
Comment (oahzxl) [inline opendit/models/latte/latte_t2v.py:1151]: maybe there is no need to determine who is conditional by:
1. set dp size = 2
2. xxx = split_sequence(xxx, group=dp_group)

File: scripts/latte/sample_dp.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
+# Adapted from Latte
```
Comment (oahzxl) [inline scripts/latte/sample_dp.py:1]: we can enable this feature by default. so maybe no need to have a new file

File: opendit/core/comm.py [hunk 3]
```diff
@@ -365,6 +365,46 @@ def gather_sequence(input_, process_group, dim, grad_scale=1.0, pad=0):
     return _GatherForwardSplitBackward.apply(input_, process_group, dim, grad_scale, pad)
 
 
+# ======================================================
```
Comment (oahzxl) [inline opendit/core/comm.py:368]: no need to add this as well

File: opendit/models/latte/latte_t2v.py [hunk 4]
```diff
@@ -1138,6 +1146,29 @@ def forward(
...
+            conditional = get_data_parallel_rank() == 0
+            (
+                hidden_states,
+                timestep,
+                encoder_hidden_states,
+                added_cond_kwargs,
+                class_labels,
+                attention_mask,
+                encoder_attention_mask,
+            ) = split_batch(
```
Comment (oahzxl) [inline opendit/models/latte/latte_t2v.py:1161]: its very good to have a batch function, but lets make it global usable. for example:
a1, a2 = batch_func(partial(split_sequence, dim=0), a1, a2)
or things like:
a1, a2 = batch_func(split_sequence, (a1, 0), (a2, 0))
or else you think better

Comment (ExtremeViscent) [reply opendit/models/latte/latte_t2v.py]: Fixed. Refer to utils.utils.batch_func()

File: scripts/latte/sample_dp.py (hunk 2 above)
Comment (ExtremeViscent) [reply scripts/latte/sample_dp.py]: Enabled in sample.py

File: opendit/models/latte/latte_t2v.py (hunk 1 above)
Comment (ExtremeViscent) [reply opendit/models/latte/latte_t2v.py:1150]: Fixed.

**Comment by ExtremeViscent** [comment]
The sequence parallel for VAE is enabled. Please kindly check it.
